src/main/java/com/crud/library/service/RentalService.java

```diff
@@ -0,0 +1,138 @@
+package com.crud.library.service;
+
+import com.crud.library.domain.Exemplar;
+import com.crud.library.domain.Rental;
+import com.crud.library.domain.User;
+import com.crud.library.dto.RentalDto;
+import com.crud.library.mapper.RentalMapper;
+import com.crud.library.repository.ExemplarRepository;
+import com.crud.library.repository.RentalRepository;
+import com.crud.library.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+
+import static com.crud.library.status.RentalStatus.*;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class RentalService {
+
+    private final RentalRepository repository;
+    private final RentalMapper mapper;
+    private final ExemplarRepository exemplarRepository;
+    private final UserRepository userRepository;
+
+    @Transactional
+    public RentalDto rentExemplar(long userId, long exemplarId, RentalDto rentalDto) {
+        if (isUserOrExemplarNotExists(userId, exemplarId)) return new RentalDto();
+        if (isExists(exemplarId)) return new RentalDto();
+        Exemplar exemplar = exemplarRepository.findById(exemplarId);
+        if (isBorrowed(exemplar)) return new RentalDto();
+        User user = userRepository.findById(userId);
+        Rental rental = mapper.mapToRental(rentalDto);
+        rental.setExemplar(exemplar);
+        rental.setUser(user);
+        exemplar.setStatus(BORROWED);
+        return mapper.mapToRentalDto(repository.save(rental));
+    }
+
+    private boolean isUserOrExemplarNotExists(long userId, long exemplarId) {
+        if (!userRepository.existsById(userId)) {
+            log.error("USER NOT EXISTS");
+            return true;
+        }
+        if (!exemplarRepository.existsById(exemplarId)) {
+            log.error("EXEMPLAR NOT EXISTS");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isExists(long exemplarId) {
+        if (repository.existsByExemplarId(exemplarId)) {
+            log.error("RENTAL ALREADY EXISTS");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isBorrowed(Exemplar exemplar) {
+        if (!exemplar.getStatus().equals(AVAILABLE)) {
+            log.error("EXEMPLAR IS NOT AVAILABLE");
+            return true;
+        }
+        return false;
+    }
+
+    @Transactional
+    public void giveBackExemplar(long id) {
```

Comment:
tutaj powinno przyjsc exemplarId oraz dodatkowe dane do zwrotu ksiazki (czy na przyklad uzytkownik nie zgubil ksiazki albo jej nie zniszczyl, czy za to zaplacil)?